### PR TITLE
3D STL Renderer

### PR DESCRIPTION
**Pull request by jvsluis**
Adds the 3d renderer in the create form
Implements resizing of canvas
Fixes some graphical bugs with the upload dropzone sizing. The dropzone now fill the whole containers height correctly

Height of dropzone fixed:
image

3d Renderer:
image

### CODE REVIEW

**Comment by Jasper-Nelligan** [comment]
Looks like there's merge conflicts that need to be resolved, particularly around the new fileCard component. Should we add the fileCard component at the top of the page above the 3D view?

Another thing I noticed is that if you select a file, click Next, and then go back, there no longer appears to be any file selected.

**Comment by jvsluis** [comment]
With the 3D Renderer I dont see a point to the file card. Maybe I'm wrong but the card would add no information. The file title, and the change button are sufficient to replace the file. Let me know what you think.

Also - I fixed the back button bug. Thx for spotting that.

**Comment by Jasper-Nelligan** [comment]
> With the 3D Renderer I dont see a point to the file card. Maybe I'm wrong but the card would add no information. The file title, and the change button are sufficient to replace the file. Let me know what you think.
> 
> Also - I fixed the back button bug. Thx for spotting that.

I guess it's not necessary anymore, but it just seems a shame to waste the fileCard. Oh well. I think you're good to remove it

File: app/src/routes/upload.js [hunk 1]
```diff
@@ -1,44 +1,172 @@
...
+        return () => {
+          console.debug("attempting to remove renderer");
+          window.removeEventListener('resize', onWindowResize, false);
+          if (containerRef.current !== null && containerRef.current.hasChildNodes()) {
+            console.debug("removed renderer");
+            containerRef.current.removeChild(renderer.domElement);
+            updateFile(null);
+          }
+        }
+      }
```
Comment (Jasper-Nelligan) [inline app/src/routes/upload.js]: Are the console logs necessary still?

**Review by Jasper-Nelligan — APPROVED** [review]

**Pull request merged**